frappe/commands/site.py

```diff
@@ -80,40 +132,63 @@ def restore(context, sql_file_path, mariadb_root_username=None, mariadb_root_pas
 			"Use `bench partial-restore` to restore a partial backup to an existing site.",
 			fg="yellow"
 		)
-		sys.exit(1)
-
-	# check if valid SQL file
-	validate_database_sql(decompressed_file_name, _raise=not force)
+		decryption_rollback(sql_file_path)
+		sys.exit(1)	
+	
+	
+	try:
+		# check if valid SQL file
+		validate_database_sql(decompressed_file_name, _raise=not force)
 
-	site = get_site(context)
-	frappe.init(site=site)
+		site = get_site(context)
+		frappe.init(site=site)
 
-	# dont allow downgrading to older versions of frappe without force
-	if not force and is_downgrade(decompressed_file_name, verbose=True):
-		warn_message = (
-			"This is not recommended and may lead to unexpected behaviour. "
-			"Do you want to continue anyway?"
-		)
-		click.confirm(warn_message, abort=True)
+		# dont allow downgrading to older versions of frappe without force
+		if not force and is_downgrade(decompressed_file_name, verbose=True):
+			warn_message = (
+				"This is not recommended and may lead to unexpected behaviour. "
+				"Do you want to continue anyway?"
+			)
+			click.confirm(warn_message, abort=True)
 
-	_new_site(frappe.conf.db_name, site, mariadb_root_username=mariadb_root_username,
-		mariadb_root_password=mariadb_root_password, admin_password=admin_password,
-		verbose=context.verbose, install_apps=install_app, source_sql=decompressed_file_name,
-		force=True, db_type=frappe.conf.db_type)
+		_new_site(frappe.conf.db_name, site, mariadb_root_username=mariadb_root_username,
+			mariadb_root_password=mariadb_root_password, admin_password=admin_password,
+			verbose=context.verbose, install_apps=install_app, source_sql=decompressed_file_name,
+			force=True, db_type=frappe.conf.db_type)
+		
+	except Exception as err:
+		print(err.args[1])
```

Comment:
Scope of the `try-except` block is too vast in this case. Reduce it to wrap only the functions where you expect the decryption Exceptions

Reply:
Not targeting the decryption error. If there is any error it should revert/rollback the changes as there is a decrypted file present (with "-enc") and for narrowing the try-except will need to add more checks for each error. Example- Invalid database SQL or wrong root password.